project/Dependencies.scala

```diff
@@ -67,6 +68,15 @@ object Dependencies {
 
   def scalaReflect(version: String) = "org.scala-lang" % "scala-reflect" % version
 
+  def spire(module: String) = Def.setting {
+    CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, 11)) => "org.spire-math" %% "spire" % "0.13.0"
+      case Some((2, 12)) => "org.spire-math" %% "spire" % "0.13.0" // 0.17.0 exists for 2.12
```

Comment:
btw the spire version (as well as some other libs versions) should be aligned with Spark versions. i.e. if it is Spark 3 than it should be `0.17.0-M1` (but probably 0.17.0 is still compatible)